### PR TITLE
Restrict agents and system administrators

### PR DESCRIPTION
**Pull request by jasonaowen**
Limit the abilities of service agents to what providers can do, and limit the abilities of system administrators to be creating, editing, and deleting users.

To test this, I logged in as `admin` (a service administrator), copied several links to a text file, and then logged in as a service agent and as a system administrator, and verified that I could not see those pages.

Note that the ops dashboard is *not* protected, and should be - that is separate from this, and I've opened #523 to track it.

Issue #10 Limit sys admins', service agents' ability to view, edit enrollments

### CODE REVIEW

File: psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java [hunk 1]
```diff
@@ -106,10 +107,8 @@
     /**
      * List of roles with full access to all profiles and tickets.
      */
-    private static final List<String> FULL_ACCESS = Arrays.asList(
-            ViewStatics.ROLE_SERVICE_AGENT,
-            ViewStatics.ROLE_SERVICE_ADMINISTRATOR,
-            ViewStatics.ROLE_SYSTEM_ADMINISTRATOR
+    private static final List<String> FULL_ACCESS = Collections.singletonList(
+            ViewStatics.ROLE_SERVICE_ADMINISTRATOR
```
Comment (cecilia-donnelly) [inline psm-app/cms-business-process/src/main/java/gov/medicaid/services/impl/ProviderEnrollmentServiceBean.java]: +1 yay!

File: psm-app/cms-web/src/main/java/gov/medicaid/controllers/ProviderDashboardController.java [hunk 2]
```diff
@@ -117,12 +118,17 @@ public ModelAndView viewDashboard() throws PortalServiceException {
...
      */
     private boolean isOperations() {
+        final Collection<String> operationsRoles = Arrays.asList(
+                ViewStatics.ROLE_SERVICE_ADMINISTRATOR,
+                ViewStatics.ROLE_SYSTEM_ADMINISTRATOR
+        );
         CMSUser user = ControllerHelper.getCurrentUser();
         String role = user.getRole().getDescription();
-        return !ViewStatics.ROLE_PROVIDER.equals(role);
+        return operationsRoles.contains(role);
```
Comment (cecilia-donnelly) [inline psm-app/cms-web/src/main/java/gov/medicaid/controllers/ProviderDashboardController.java]: Great idea!

File: psm-app/cms-web/WebContent/WEB-INF/spring-security.xml [hunk 3]
```diff
@@ -15,10 +15,10 @@
...
   <s:http entry-point-ref="authenticationProcessingFilterEntryPoint">
     <s:intercept-url pattern="/system/**" access="ROLE_SYSTEM_ADMINISTRATOR"/>
-    <s:intercept-url pattern="/admin/**" access="ROLE_SERVICE_ADMINISTRATOR,ROLE_SERVICE_AGENT,ROLE_SYSTEM_ADMINISTRATOR"/>
-    <s:intercept-url pattern="/agent/**" access="ROLE_SERVICE_ADMINISTRATOR,ROLE_SERVICE_AGENT,ROLE_SYSTEM_ADMINISTRATOR"/>
-    <s:intercept-url pattern="/provider/profile/edit" access="ROLE_PROVIDER,ROLE_SERVICE_AGENT,ROLE_SERVICE_ADMINISTRATOR"/>
-    <s:intercept-url pattern="/provider/profile/renew" access="ROLE_PROVIDER,ROLE_SERVICE_AGENT,ROLE_SERVICE_ADMINISTRATOR"/>
+    <s:intercept-url pattern="/admin/**" access="ROLE_SERVICE_ADMINISTRATOR"/>
+    <s:intercept-url pattern="/agent/**" access="ROLE_SERVICE_ADMINISTRATOR"/>
+    <s:intercept-url pattern="/provider/profile/edit" access="ROLE_PROVIDER,ROLE_SERVICE_ADMINISTRATOR"/>
+    <s:intercept-url pattern="/provider/profile/renew" access="ROLE_PROVIDER,ROLE_SERVICE_ADMINISTRATOR"/>
```
Comment (cecilia-donnelly) [inline psm-app/cms-web/WebContent/WEB-INF/spring-security.xml]: Does this mean that service admins can change the provider's account information?  Or does "profile" here mean something different?

**Comment by cecilia-donnelly** [comment]
@jasonaowen, the error pages on this branch aren't styled (noticed when I created a new "agent" user without the mail server running).  Presumably it just hasn't been rebased onto master, so doesn't have the fix from #505, but figured I'd mention it.

**Comment by jasonaowen** [comment]
@cecilia-donnelly yep, that's exactly why! I can rebase onto master, if you prefer, but I think showing even a poorly formatted error page counts as success for this PR.

**Comment by cecilia-donnelly** [comment]
@jasonaowen no need!  Agreed, I think we're in good shape here. :)

File: psm-app/cms-web/WebContent/WEB-INF/spring-security.xml (hunk 3 above)
Comment (jasonaowen) [reply psm-app/cms-web/WebContent/WEB-INF/spring-security.xml]: Good question! That took a little bit of digging.

"Profile" here is similar to an approved enrollment. If you log in as a provider with at least one approved enrollment, you can click on the "My Profile" tab. From there, you can view your approved enrollment, edit it (which resubmits it with a request type of "Update"), or renew it (which resubmits it with a request type of "Renew").

It turns out that for admins, the `Edit` and `Renew` links in the `Approved Enrollments` tab of the enrollments list go to those two URLs, which is why admins have this access.

Comment (cecilia-donnelly) [reply psm-app/cms-web/WebContent/WEB-INF/spring-security.xml]: Thanks for the explanation!

**Pull request merged**